aten/src/ATen/native/LinearAlgebra.cpp

```diff
@@ -1892,36 +1935,146 @@ TORCH_IMPL_FUNC(bmm_out_cpu)
     }
 }
 
-Tensor& dot_out(const Tensor& self, const Tensor& other, Tensor& result) {
-  auto output_device = result.device();
-  auto input1_device = self.device();
-  auto input2_device = other.device();
-  // check if the input & output tensors are on the same device.
-  TORCH_CHECK(
-    (output_device == input1_device) && (input1_device == input2_device),
-    "dot: Expected the output and input tensors to be on the "
-    "same device, but got the output tensor on ", output_device,
-    ", the 'input' tensor on ", input1_device, ", and the 'other' tensor on ", input2_device);
-  at::native::resize_output(result, {});
-  TORCH_CHECK(result.scalar_type() == self.scalar_type(),
-           "result dtype ", result.scalar_type(), " does not match input dtype ", self.scalar_type());
-  return result.fill_(self.dot(other));
-}
-
-Tensor& vdot_out(const Tensor& self, const Tensor& other, Tensor& result) {
-  auto output_device = result.device();
-  auto input1_device = self.device();
-  auto input2_device = other.device();
-  // check if the input & output tensors are on the same device.
-  TORCH_CHECK(
-    (output_device == input1_device) && (input1_device == input2_device),
-    "vdot: Expected the output and input tensors to be on the "
-    "same device, but got the output tensor on ", output_device,
-    ", the 'input' tensor on ", input1_device, ", and the 'other' tensor on ", input2_device);
-  at::native::resize_output(result, {});
-  TORCH_CHECK(result.scalar_type() == self.scalar_type(),
-           "result dtype ", result.scalar_type(), " does not match input dtype ", self.scalar_type());
-  return result.fill_(self.vdot(other));
+void vdot_out_impl(
+    const Tensor& self,
+    const Tensor& other,
+    const Tensor& result);
+
+template <typename scalar_t>
+scalar_t dot_impl(
+    int64_t n,
+    scalar_t* x,
+    int64_t incx,
+    scalar_t* y,
+    int64_t incy);
+
+template <typename scalar_t>
+scalar_t vdot_impl(
+    int64_t n,
+    scalar_t* x,
+    int64_t incx,
+    scalar_t* y,
+    int64_t incy);
+
+void dot_out_impl(
+    const Tensor& self,
+    const Tensor& other,
+    const Tensor& result) {
+    auto output_device = result.device();
+    auto input1_device = self.device();
+    auto input2_device = other.device();
+    // check if the input & output tensors are on the same device.
+    TORCH_CHECK(
+        (output_device == input1_device) && (input1_device == input2_device),
+        "vdot: Expected the output and input tensors to be on the "
+        "same device, but got the output tensor on ",
+        output_device,
+        ", the 'input' tensor on ",
+        input1_device,
+        ", and the 'other' tensor on ",
+        input2_device);
+    if (self.is_complex()) {
+    if (self.is_conj()) {
+      if (other.is_conj()) {
+        Tensor temp_result = at::empty({}, result.options());
+        dot_out_impl(self.conj(), other.conj(), temp_result);
+        result.copy_(temp_result.conj());
```

Comment:
I'm curious - why is it a `copy_` here and not a `fill_` (like used in some other places)?